cabal.project

```diff
@@ -335,6 +337,31 @@ source-repository-package
 allow-older: *
 allow-newer: *
```

Comment:
Why are these needed?

Reply:
1. Because ppl don't maintain .cabal files properly anymore since stack/nix/cabal-freeze so the version bounds are often wrong and lead to resolution failures
2. our `cardano-<ver>.yaml` snapshots often specify stuff that doesn't resolve and doesn't actually build, but isn't used by any package. E.g. input-output-hk/cardano-sl-x509 doesn't build with lts-14.25... try to add it to cardano-wallet and see it [blow up](https://gist.github.com/hasufell/17854ddfc9aa14897b6bb6e10069129b)... I fixed it [here](https://github.com/input-output-hk/cardano-sl-x509/commit/bf7e67c8af78d48453a12bb2ada85325fe7d26c9). However, cabal doesn't lazily evaluate the build plan like stack and doesn't ignore .cabal version bounds like stack.

So... you get all sorts of resolution errors, but when you ignore them, the build passes.